source/agora/consensus/EnrollmentManager.d

```diff
@@ -943,7 +944,9 @@ unittest
     assert(man.addValidator(enroll, 10, &storage.findUTXO));
     assert(!man.getPreimage(10, preimage));
     assert(man.getPreimage(11, preimage));
+    assert(preimage.hash == man.cycle.preimages[$ - 1]);
```

Comment:
It would be better if we had tests about the interface of this function rather than the implementation.
The previous behavior was completely broken, and the code introduced should catch that without a doubt, but at the moment I'm not sure it does that. We really need more network test for pre-image and Enrollments.

Reply:
The test code is incomplete. I will add more test code for pre-image and enrollments in the agora.test.EnrollmentManager module.